lib/LWP/MediaTypes.pm

```diff
@@ -48,9 +48,15 @@ sub guess_media_type
 
     my $fullname;
     if (ref($file)) {
-	# assume URI object
-	$file = $file->path;
-	#XXX should handle non http:, file: or ftp: URIs differently
+        if ($file->can('path')) {
```

Comment:
@genio pointed out that we should probably check that this is `blessed` before we check `can`.